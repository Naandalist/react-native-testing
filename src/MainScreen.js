import React from 'react';
import {StyleSheet, Text, View} from 'react-native';

export default function mainScreen() {
  return (
    <View style={styles.container}>
      <Text>Test UI Renderer</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'blue',
    flex: 1,
  },
});
