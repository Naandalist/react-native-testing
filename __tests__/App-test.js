import React from 'react';
import {create} from 'react-test-renderer';
import MainScreen from '../src/MainScreen';

const tree = create(<MainScreen />);

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});

/**
 * this will create new file snapshot to compare in the future. 
 * Then, we use 'yarn test -u' toupdate the snapshot
 */