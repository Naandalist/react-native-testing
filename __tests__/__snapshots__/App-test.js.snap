// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "red",
      "flex": 1,
    }
  }
>
  <Text>
    Test UI Renderer
  </Text>
</View>
`;
